tests/v2/validation/deleting/delete_init_node.go

```diff
@@ -0,0 +1,30 @@
+package deleting
+
+import (
+	"testing"
+
+	"github.com/rancher/shepherd/clients/rancher"
+	"github.com/rancher/shepherd/extensions/clusters"
+	"github.com/rancher/shepherd/extensions/nodes"
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/require"
+)
+
+func deleteInitNode(t *testing.T, client *rancher.Client, clusterID string) {
+	initNode, err := nodes.GetInitNode(client, clusterID)
+	require.NoError(t, err)
+
+	err = nodes.DeleteNodeRKE2K3S(client, initNode)
+	require.NoError(t, err)
+
+	logrus.Info("Awaiting machine replacement...")
+
+	err = clusters.WaitClusterToBeUpgraded(client, clusterID)
```

Comment:
nit - can you leave a comment as to why there is a need to have this called twice? We've talked offline of the need, but it will be good to have this documented.